test/test_folders.py

```diff
@@ -0,0 +1,134 @@
+import unittest
+import os.path
+
+from tableaudocumentapi import Datasource
+import xml.etree.ElementTree as ET
+
+
+TEST_ASSET_DIR = os.path.join(
+    os.path.dirname(__file__),
+    'assets'
+)
+TEST_TDS_FILE = os.path.join(
+    TEST_ASSET_DIR,
+    'folder_test.tds'
+)
+
+
+class TestFieldChange(unittest.TestCase):
+
+    def setUp(self):
+        self.tds = Datasource.from_file(TEST_TDS_FILE)
+
+    def current_hash(self):
+        """ Return a hash of the current state of the XML.
+
+        Allows us to easily identify whether the underlying XML-structure
+        of a TDS-file has actually changed. Avoids false positives if,
+        for example, a fields value has changed but the XML hasn't.
+        """
+        return hash(ET.tostring(self.tds._datasourceTree.getroot()))
```

Comment:
Go ahead and stick with regular tempfile based tests for now. I don't think I will have time to implement the file-object stuff for a while (the main issue is providing a filename for save/save_as).